packages/runed/src/lib/utilities/PersistedState/PersistedState.svelte.ts

```diff
@@ -0,0 +1,198 @@
+import { untrack } from "svelte";
+import { BROWSER, DEV } from "esm-env";
+import { addEventListener } from "$lib/internal/utils/event.js";
+
+type Serializer<T> = {
+	serialize: (value: T) => string;
+	deserialize: (value: string) => T;
+};
+
+type GetItemResult<T> =
+	| {
+			found: false;
+			value: null;
+	  }
+	| {
+			found: true;
+			value: T;
+	  };
+
+interface StorageAdapter<T> {
```

Comment:
I think this type should be exported since it is part of the public options.

Also, what about making the contract asynchronous?  This way one could write a storage adapter for [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API).

Reply:
@webJose totally agree. This PR is a bit old as we were dealing with a bunch of rapid Svelte changes at the time. I'd like to be able to support generic storage adapters and asynchronous contracts to support IndexedDB or any other storage solution one could think of.